state/execution.go

```diff
@@ -339,19 +340,22 @@ func updateValidators(currentSet *types.ValidatorSet, abciUpdates []abci.Validat
 			// remove val
 			_, removed := currentSet.Remove(address)
 			if !removed {
-				return fmt.Errorf("Failed to remove validator %X", address)
+				return fmt.Errorf("failed to remove validator %X", address)
 			}
 		} else if val == nil {
-			// add val
+			// Set Accum to -totalVotingPower to make sure validators can't unbond/rebond to reset their (potentially
```

Comment:
could you please link the documentation issue (create one if needed)?

Reply:
There already is an issue about the general proposer selection algo spec: https://github.com/tendermint/tendermint/issues/1558
Why would we link it in the code, though?

Reply:
To make it discoverable when you read the code.